src/MDMTransaction.h

```diff
@@ -27,9 +27,36 @@ NS_SWIFT_NAME(Transaction)
 
 #pragma mark Adding plans to a transaction
 
-/** Associate an plan with a given target. */
+/**
+Associates a plan with a given target.
+
+@param plan The plan to add to this transaction.
+@param target The target on which the plan can operate.
+*/
+- (void)addPlan:(nonnull id<MDMPlan>)plan
+       toTarget:(nonnull id)target
+  NS_SWIFT_NAME(add(plan:to:));
+
+/**
+ Associates a named plan with a given target.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan.
+ */
 - (void)addPlan:(nonnull id<MDMPlan>)plan
        toTarget:(nonnull id)target
-    NS_SWIFT_NAME(add(plan:to:));
+       withName:(nonnull NSString *)name
+  NS_SWIFT_NAME(add(plan:to:withName:));
+
+/**
+ Removes a named plan from the transaction.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan
+ */
+- (void)removePlanNamed:(nonnull NSString *)name
+  NS_SWIFT_NAME(remove(name:));
```

Comment:
https://github.com/material-motion/material-motion-runtime-objc/pull/88/commits/c47d5309b615b48bb2b6bf5d925a42dbbd1d4a70 has some thoughts on where this _could_ go, but I'm not sure I've fully grasped the intent of adding a `removePlan` method to the `MDMPlanPerforming` protocol.

Specifically, I'm not sure creating an instance of the performer when removing a plan is correct.

I guess one alternative to this would be to alter the `MDMPlan` protocol to look something like:
```
@protocol MDMPlan <NSObject, NSCopying>

- (nonnull Class)performerClass;

- (nullable id<MDMPerforming>)performer;

@end
```
Otherwise, if we need access to the original performer instance, we could allow access via `MDMPerformerGroup` ?